include/dr/shp.hpp

```diff
@@ -4,6 +4,7 @@
 
 #pragma once
 
+#include <dr/detail/ranges_shim.hpp>
```

Comment:
`details` and `util` are getting back to this file :) can we keep them included in some second-order includes rather than in main shp.hpp?

unless they are really meant to be included by user and used directly by user

Reply:
Fair point, will remove.